libs/agno/agno/agent/agent.py

```diff
@@ -5867,6 +5882,35 @@ async def _aread_or_create_session(
 
         return agent_session
 
+    def _validate_session(
```

Comment:
I don't see why this is necessary. You only need to make sure wherever we access the session history that you also filter for the agent_id, which is sensible. Converting this and adding another object seems unnecessarily costly